### PR TITLE
Update readme templates and generation

### PR DESCRIPTION
**Pull request by joeyfreund**
# What?

First step towards https://github.com/Shopify/argo-admin-private/issues/1469 

 1. Update the README to [only include instructions for generating the starter code](https://github.com/Shopify/argo-admin-template/blob/jf/update-readme-generation/README.md)
 2. Add readme-templates for each extension type (e.g. here's [the template for product subscription extensions](https://github.com/Shopify/argo-admin-template/blob/jf/update-readme-generation/scripts/generate/templates/PRODUCT_SUBSCRIPTION/README.template.md))
 3. Modify the `generate` script to overwrite `README.md` with the correct readme-template.

In follow-up PR's, we can improve the context of each specific template.       
For example, see the README for [product subscription extensions](https://github.com/Shopify/argo-admin-template/blob/jf/update-readme-generation/scripts/generate/templates/PRODUCT_SUBSCRIPTION/README.template.md), it's not even clear what to do once we run `npm server` (what is this little web app running in localhost? What am I support to do with it?).

# Why?

The current README starts by making a distinction between the case where someone clones the repo and the case where the Shopify CLI clones it for them.

 - If the user cloned the repo directly, then there's no point  in mentioning the CLI
 - If the user used the CLI (I'm assuming the CLI runs `generate`), then there's no point in going into the details of how to manually `generate` the starter code

# Question for reviewers

> I'm assuming the CLI runs `generate`

Is this assumption correct?

# How to 🎩 ?

```sh
cd /tmp
git clone git@github.com:Shopify/argo-admin-template.git my_extension
cd my_extension
git checkout jf/update-readme-generation
npm install
npm run generate -- --type=PRODUCT_SUBSCRIPTION
```

When asked to choose a template, select any option (it doesn't matter which one), and verify that `README.md` contains the expected content:

```sh
cat README.md
# The title of the readme should be `Production Subscription app extension`
```

### CODE REVIEW

**Review by henrytao-me — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,49 +1,32 @@
-# Setup
+# Shopify Admin, app extensions - Starter code
 
-## CLI
-If you have grabbed this template from the Shopify CLI you should be good to go!
+This repo contains starter code for apps that use _Argo_ to extend the UI of the _Shopify Admin_.       
+For more about _Argo_ and admin extensions, please [read our docs](https://shopify.dev/tutorials/product-subscription-extension-overview).
 
-Start developing your awesome extension.
+## Getting started
```
Comment (iainmcampbell) [inline README.md:6]: We should encourage people to use the CLI before manually cloning this repo: https://shopify.dev/tutorials/getting-started-product-subscription-extension

File: README.md [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
-### Build
+_**Note:** Currently `PRODUCT_SUBSCRIPTION` is the only supported extension type._
 
-During the setup, a new script has been added to your `package.json` for building your code. Run the command to start developing locally. See [Build with argo-admin-cli](https://www.npmjs.com/package/@shopify/argo-admin-cli#build).
+## What's next?
 
-```bash
-  npm run build
-```
+Check out the generated readme and index files (at the root of this repo), they contain further instructions and code to get you started.
```
Comment (iainmcampbell) [inline README.md:32]:
```suggestion
The `generate` command has created new files at the root of this repo. Check out the new `README.md` for further instructions, and `index.js/ts/tsx` for starter code.```

File: scripts/generate/templates/PRODUCT_SUBSCRIPTION/README.template.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+For more info, [read the docs](https://shopify.dev/tutorials/product-subscription-extension-overview).
+
+
+### Local development
+
+Run a local server:
+
+```bash
+  npm run server
```
[the remark this answers is not in the document]
Comment (joeyfreund) [reply scripts/generate/templates/PRODUCT_SUBSCRIPTION/README.template.md:13]: @vividviolet , just double checking ... What's the yarn command? Is it `yarn server`?

[the remark this answers is not in the document]
Comment (vividviolet) [reply scripts/generate/templates/PRODUCT_SUBSCRIPTION/README.template.md:13]: yup! `yarn server`

**Comment by joeyfreund** [comment]
@vividviolet and @iainmcampbell , do mind giving this another look?

Trish, I addressed all of you comments 🙏 
(at least I think I did, please double check).

Iain, I re-worded the README, please take a look and tell me what you think.

File: README.md [hunk 4]
```diff
@@ -1,49 +1,44 @@
...
 
-#### npm
-```bash
-## With npm
-npm install
+**Note:** There is no need for you to clone this repo directly.
+The CLI will take care of cloning the repo and generating all required files.
 
-npm run generate -- --type=PRODUCT_SUBSCRIPTION
+## Contributing
```
Comment (iainmcampbell) [inline README.md:14]: Not sure about this one, do we expect 3p contributions here?

We also seem to have lost the 'Getting started' heading.

File: README.md [hunk 5]
```diff
@@ -1,49 +1,44 @@
...
-## From github
-If you grabbed this template directly from github:
+[Use the Shopify App CLI to create your _Argo_-enabled extension](https://shopify.dev/tutorials/getting-started-product-subscription-extension#scaffold-a-product-subscription-app-extension).
 
-#### npm
-```bash
-## With npm
-npm install
+**Note:** There is no need for you to clone this repo directly.
+The CLI will take care of cloning the repo and generating all required files.
```
Comment (iainmcampbell) [inline README.md:12]:
```suggestion
**Note:** We recommend [using the Shopify App CLI to create your Argo-enabled extension](https://shopify.dev/tutorials/getting-started-product-subscription-extension#scaffold-a-product-subscription-app-extension).

If you clone this repo directly and follow the instructions below, you’ll need to set up the Shopify App CLI manually to use your extension.
```

**Review by iainmcampbell — APPROVED** [review]

**Pull request merged**